functions.php

```diff
@@ -225,6 +225,38 @@ function enqueue_scripts() {
 		);
 		wp_localize_script( 'cc_base_script', 'Ajax', $ajax_data );
 	}
+
+	/**
+	 * Given than WordPress generates css class names following the convention:
+	 * .has-{slug}-color and .has-{slug}-background-color. 
+	 * When rendering pages on the frontend, these are the css classes applied to the different HTML elements
+	 * which is different from how the Vocabulary package generates it's CSS classes:
+	 * .has-color-{slug} and .has-backgound-{slug}. 
+	 * This function below takes the colors we have added to 
+	 * the color pallete of wordpress and generates WordPress compatible class names. 
+	 * The styles generated are then added to the HTML pages as Internal CSS styles. 
+	 * We do this by using the Wordpress action (wp_head) which lets us run this function when the head is called.
+	 * @var editor-color-pallete
+	 */
+	function custom_css()
+	{
+	   $palette = get_theme_support( 'editor-color-palette' );
+	   if( !$palette ) { return; } // If our color pallete has not been define don't run this function.
+		   
+	   // format styles
+	   $styles = ":root .has-background { background-color: var(--bgColor); }
+	   :root .has-text-color { color: var(--textColor); } :root .has-inline-color { color: var(--textColor); } ";
+	   foreach( $palette[0] as $name => $value ) {
+		   $slug = $value['slug'];
+		   $color = $value['color'];
+	   
+		   $styles .= ".has-{$slug}-background-color { --bgColor: {$color}; } ";
+		   $styles .= ".has-{$slug}-color { --textColor: {$color}; } ";
+	   }
```

Comment:
Double-check the indentation in this code block. If possible, use a code linter or formatted to make indentation issues more clear in your editor.